### ISSUE TITLE
Firebase Messaging [FIRApp registerAsConfigurable:]: unrecognized selector sent to class 0x11034d9e8

### ISSUE THREAD

**Issue by coosmiyn** [xamarin/GoogleApisForiOSComponents#311]
I get this error in the console when I try to deploy the app, tried both on simulator and physical device. It displays the splash image then it backgrounds the app. No exception is thrown, the debugger just disconnects.

It seems that this only happens when I use Firebase Messaging. If I delete any of the code related to Firebase Messaging (but not deleting the NuGet) it works as intended.

I am using Visual Studio for mac, the ios 13 sdk and the latest Firebase NuGets. Tried downgrading but nothing helps.

```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '+[FIRApp registerAsConfigurable:]: unrecognized selector sent to class 0x11034d9e8'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23b98bde __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff503b5b20 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff23bb9604 +[NSObject(NSObject) doesNotRecognizeSelector:] + 132
	3   CoreFoundation                      0x00007fff23b9d7bc ___forwarding___ + 1436
	4   CoreFoundation                      0x00007fff23b9f6c8 _CF_forwarding_prep_0 + 120
	5   libobjc.A.dylib                     0x00007fff503bcdf0 load_images + 1226
	6   ???                                 0x0000000110c24d79 0x0 + 4576136569
	7   ???                                 0x0000000110c31970 0x0 + 4576188784
	8   ???                                 0x0000000110c30786 0x0 + 4576184198
	9   ???                                 0x000<…>
```

**Comment by coosmiyn** [comment]
Tried it again today, but I get another error now:

```
Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Description: DYLD, dependent dylib '@rpath/protobuf.framework/protobuf' not found for '/private/var/containers/Bundle/Application/DA138C0F-1292-4C17-A793-0443C743F7F6/ClubHub.iOS.app/Frameworks/FirebaseMessaging.framework/FirebaseMessaging', tried but didn't find: '/private/var/containers/Bundle/Application/DA138C0F-1292-4C17-A793-0443C743F7F6/ClubHub.iOS.app/Frameworks/protobuf.framework/protobuf' '/private/var/containers/Bundle/Application/DA138C0F-1292-4C17-A793-0443C743F7F6/ClubHub.iOS.app/Frameworks/FirebaseMessaging.framework/Frameworks/protobuf.framework/protobuf' '/private/var/containers/Bundle/Application/DA138C0F-1292-4C17-A793-0443C743F7F6/ClubHub.iOS.app/Frameworks/protobuf.framework/protobuf' '@rpath/protobuf.framework/protobuf' '/System/Library/Frameworks/protobuf.framework/protobuf'
Highlighted by Thread:  0
```

Still, it happens just when I use CloudMessaging.

**Comment by Adlorem** [comment]
Got the same problem with latest packages for iOS Firebase Messaging. The app starts but nothing happens past launch screen. There are no errors no other info. Downgrading doesn't help at all. I am not able to compile project with new XCode 11.1 and iOS 13.1.

**Comment by zzm388** [comment]
Same problem here using latest firebase messaging for IOS. I'm building my app on VS for windows. App displays splash screen and then disappears. Device log shows message about missing protobuf library as @coosmiyn reports.

**Comment by remeesms** [comment]
Same Issue here, Any update on this?

**Comment by SotoiGhost** [comment]
After a long search, we found out the root of the issue. The issue is that Firebase.CloudMessaging was built against Protobuf v3.10 due to Podspec version rules and Firebase.Core against v3.8, between these versions, the **Module name** changed from `Protobuf` to `protobuf`. Therefore, F.Core has `Protobuf.framework` embedded and F.CloudMessaging is looking for `protobuf.framework`.

Downgrading CloudMessaging to v3.1.2 could be a workaround but could bring unexpected behaviors. We are working on a fix right now.

**Comment by moustafashaban** [comment]
@SotoiGhost great news, 
Please add a comment here once it is done, thanks in advanced

**Comment by Tokta** [comment]
Hello, I have the same issue, but only on my CI machine, on my machine compile correctly. I was thinking about caching on my laptop but by deintegrating pods and reinstall them all it keeps working.

**Comment by doudouLin** [comment]
have the same problem, application crashes when start debuging. is there solution? tried downgrade cloudmessaging to 3.1.2, but can not compile anymore

**Comment by SotoiGhost** [comment]
@doudouLin The fix was merge and now it's building on Azure. Hope today the fix can be released to NuGet

**Comment by doudouLin** [comment]
@SotoiGhost thank you very much for the quick reply! was struggling for hours, hihi.

**Comment by SotoiGhost** [comment]
Guys! Letting you know that Firebase Cloud Messaging v4.1.1.1 has been just pushed to NuGet. Just wait a bit so it can be indexed. If something happens related to this, please, fill a new issue. Best!

**Comment by SotoiGhost** [comment]
And FYI, there's an [issue with NuGet][1] right now where dependencies **.targets** and **.props** are not being referenced correctly. If you face an error similar to issue #318, please, add [`Xamarin.Firebase.iOS.CloudMessaging`][2] dependencies explicitly to your app. Hope this helps! Best!

[1]: https://github.com/NuGet/Home/issues/8694
[2]: https://www.nuget.org/packages/Xamarin.Firebase.iOS.CloudMessaging/

**Comment by remeesms** [comment]
@SotoiGhost  Thanks!!, Seems it's working with the updated version.

**Comment by remeesms** [comment]
@SotoiGhost , After updating  Xamarin.Firebase.iOS.CloudMessaging to v4.1.1.1 the notifications are not receiving. Is anyone try to update the version. Only change in the code is getting the token as shown below 
I```
nstanceId.SharedInstance.GetInstanceId((result, error) =>
            {
                if (error != null)
                {
                   Log.Debug("GetInstanceId error", error.LocalizedDescription);
                    
                }
                else
                {
                    PushNotificationService.RegisterDevice(result.Token);
                }
            }); 
```

**Comment by jaspervanmegroot** [comment]
Its stills crash at startup..

**Comment by remeesms** [comment]
While using  SignInWithCredential the app is crashing with "Specified cast is not valid " error. 
Its happening after updating the firebase auth to version="6.2.1".

Firebase versions using:
```
  <package id="Xamarin.Firebase.iOS.Analytics" version="5.1.4.1" targetFramework="xamarinios10" />
  <package id="Xamarin.Firebase.iOS.Auth" version="6.2.1" targetFramework="xamarinios10" />
  <package id="Xamarin.Firebase.iOS.CloudMessaging" version="4.1.1.1" targetFramework="xamarinios10" />
  <package id="Xamarin.Firebase.iOS.Core" version="6.1.0" targetFramework="xamarinios10" />
  <package id="Xamarin.Firebase.iOS.Database" version="6.0.0" targetFramework="xamarinios10" />
  <package id="Xamarin.Firebase.iOS.InstanceID" version="4.2.1" targetFramework="xamarinios10" />
  <package id="Xamarin.Firebase.iOS.Storage" version="3.4.0" targetFramework="xamarinios10" />
```

Code Sample:
```
var credendtials = PhoneAuthProvider.DefaultInstance.GetCredential(verificationId, verificationCode);
Auth.DefaultInstance.SignInWithCredential(credential: credendtials, (task, error) =>
            {
                if (error != null)
                {                  
                    tcsPhone.SetResult(false);
                }
                else
                {
                    tcsPhone.SetResult(true);
                }

            });
```
Stack Trace error:

>  at (wrapper castclass) System.Object.__castclass_with_cache(object,intptr,intptr)\n  at ObjCRuntime.Runtime.ConstructNSObject[T] (System.IntPtr ptr, System.Type type, ObjCRuntime.Runtime+MissingCtorResolution missingCtorResolution) [0x00050] in <d5ac8b1069644e148a96bf0a4b22f2fd>:0 \n  at ObjCRuntime.Runtime.GetNSObject[T] (System.IntPtr ptr) [0x000d1] in <d5ac8b1069644e148a96bf0a4b22f2fd>:0 \n  at ObjCRuntime.Trampolines+SDAuthResultHandler.Invoke (System.IntPtr block, System.IntPtr user, System.IntPtr error) [0x00016] in <d7f3560f732b4e54aa3981a41bb500be>:0 \n  at (wrapper native-to-managed) ObjCRuntime.Trampolines+SDAuthResultHandler.Invoke(intptr,intptr,intptr)\n  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)\n  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.4.0.2/src/Xamarin.iOS/UIKit/UIApplication.cs:86 \n  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0000e] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.4.0.2/src/Xamarin.iOS/UIKit/UIApplication.cs:65 \n  at Myapp.Touch.Application.Main (System.String[] args) [0x00001] in /Users/developeradmin/Myapp/Alpha8/Mobile/Sources/Myapp.Touch/Main.cs:9

**Comment by vincooys** [comment]
I'm having this issue now, and not able to pinpoint where it's coming from. Getting the error from device log:

@rpath/Protobuf.framework/Protobuf not found for /private/var/containers/Bundle/Application/3343439-jflkd-34343--/APPNAME/Frameworks/FirebaseFirestore.framework/FirebaseFirestore, tried but didn't find: '/private/var/containers/Bundle/Application/434343-2ADD-3433-543333D22/APPNAME/Frameworks/Protobuf.framework/Protobuf'

I have the following packages installed in my Xamarin.iOS project:

Xamarin.Firebase.iOS.Analytics (6.2.1)
Xamarin.Firebase.iOS.Auth (6.4.2)
Xamarin.Firebase.iOS.CloudFirestore (1.4.2.2)
Xamarin.Firebase.iOS.CloudMessaging (4.1.1.2)
Xamarin.Firebase.iOS.Core (6.6.0)
Xamarin.Firebase.iOS.Crashlytics (3.14.0.3)
Xamarin.Firebase.iOS.InstanceID (4.3.0)
Xamarin.Firebase.iOS.Storage (3.4.0.1)


I've tried delete every folder I can think of (XamarinBuildCache, obj, bin, etc). Rolling back versions of packages to previous versions...nothing I do stop my device from crashing on startup and displaying this error in the device log :(

It all started when I installed the CloudMessaging package. Everything else works fine.

**Comment by edooley** [comment]
@SotoiGhost I am having the same issue as @vincooys. My Nugets and versions are the same as his except I am not referencing Crashlytics but AM referencing Xamarin.Firebase.iOS.DynamicLinks (4.0.1.1)

**Comment by AmandaU** [comment]
I am having the same issue as @vincooys  and @edooley when upgrading to Xamarin.Firebase.iOS.CloudMessaging v4.1.1.2 and Xamarin.Firebase.iOS.Analytics v6.2.1.
 Downgrading CloudMessaging to v4.1.1.1 and Analytics v6.0.4.1 solved the problem. Both packages must be downgraded

**Comment by raindeer** [comment]
Still having the same problem..

**Comment by randomroxks** [comment]
Same here. 
![Screen Shot 2020-03-18 at 12 16 38 PM](https://user-images.githubusercontent.com/35347402/76988215-5e430600-6912-11ea-8329-4586180714ec.png)

**Comment by Kingram91** [comment]
This issue needs to be reopened, as the Problem still Exists. 
For me, the problem started to occur once I added Xamarin.Google.iOS.TagManager nuget

**Comment by mindphaser** [comment]
I have the same issue with these versions:
Xamarin.Firebase.iOS.CloudMessaging (4.1.1.2)
Xamarin.Firebase.iOS.Core (6.6.0)
Xamarin.Firebase.iOS.InstanceID (4.3.0)

It works only after downgrading CloudMessaging to 3.1.2. All working versions, respectively are:
Xamarin.Firebase.iOS.CloudMessaging (3.1.2)
Xamarin.Firebase.iOS.Core (5.1.3)
Xamarin.Firebase.iOS.InstanceID (3.2.1)

**Comment by MappingSteve** [comment]
@SotoiGhost - this crash still occurs with Xamarin.Firebase.iOS.CloudMessaging (4.3.0) + .Core (6.6.6) + .InstanceID (4.3.3).

>dyld: dependent dylib '@rpath/Protobuf.framework/Protobuf' not found for '/private/var/containers/Bundle/Application/...
  /FirebaseMessaging.framework/FirebaseMessaging',  
tried but didn't find:  
'/private/var/containers/Bundle/Application/... /Protobuf.framework/Protobuf'

**Comment by jastorgac** [comment]
I'm having this issue with: 

Xamarin.Firebase.iOS.Core (6.6.6)
Xamarin.Firebase.iOS.DynamicLinks (4.0.8)
Xamarin.Firebase.iOS.RemoteConfig (4.4.9)

**Comment by udiidan** [comment]
Did anyone find a solution for this issue? 
I can run the app after deleting Obj, Bin and  XamarinBuildCache. But after making any change in the code, it crashes again.
I can't really understand how this issue is closed as clearly it's not fixed.

**Comment by chrisfoulds** [comment]
I have the same issue @SotoiGhost this issue is still ongoing

**Comment by chrisfoulds** [comment]
For reference it only occurs if Xamarin.iOS framework is newer than 13.16.0.13
If I downgrade to that is all works.
It is like Xamarin.Build.Download is not working correctly with the latest framework

**Comment by volcano619** [comment]
This is happening even in the latest Release

**Comment by rayxxi** [comment]
Did anyone find a solution for this issue?

**Comment by udiidan** [comment]
> Did anyone find a solution for this issue?

Currently it's working for me:
Xcode Version 12.4

Visual Studio Community 2019 for Mac
Version 8.8.10 (build 2)

Xamarin.iOS
Version: 14.10.0.4

Xamarin.Firebase.iOS.CloudMessaging (4.7.1)
Xamarin.Firebase.iOS.Core (6.10.4)
Xamarin.Firebase.iOS.InstanceID (4.8.0)

**Comment by rayxxi** [comment]
I am using VS Version 8.9.2 (build 0), I'll try to downgrade to 8.8.10
